Library/Homebrew/sandbox.rb

```diff
@@ -68,6 +69,11 @@ def allow_write_cellar(formula)
     allow_write_path formula.var
   end
 
+  # Xcode projects expect access to certain cache/archive dirs.
+  def allow_write_xcode
+    allow_write_path "/Users/#{ENV["USER"]}/Library/Developer/Xcode/DerivedData/"
```

Comment:
Ah, problem:

```ruby
  def xcodebuild(*args)
    removed = ENV.remove_cc_etc
    system "xcodebuild", *args, "-derivedDataPath", "#{buildpath}/.brew_home/DerivedData"
  ensure
    ENV.update(removed)
  end
```

```
xcodebuild: error: The flag -scheme is required when specifying -derivedDataPath.
```

There are only two formula which use the `-scheme` flag, and that flag isn't applicable to every project.